code/weapon/weapons.cpp

```diff
@@ -1480,7 +1480,33 @@ int parse_weapon(int subtype, bool replace, const char *filename)
 		}
 	}
 
-
+		// set the number of missles or ballistic ammo reloaded per batch
+	int reloaded_each_batch;
+	
+	if (optional_string("$Rearm Ammo Increment:"))
+	{
+		stuff_int( &reloaded_each_batch );
+
+		if (reloaded_each_batch > 0) {
+			wip->reloaded_per_batch = reloaded_each_batch;
+		} else	{
+			if (subtype == WP_MISSILE) {
+				wip->reloaded_per_batch = REARM_NUM_MISSILES_PER_BATCH;
+				Warning(LOCATION, "Rearm batch amount of 0 or less in weapon %s, setting to the default 4 for missiles.", wip->name);
+			} else {
+				wip->reloaded_per_batch = REARM_NUM_BALLISTIC_PRIMARIES_PER_BATCH;
+				Warning(LOCATION, "Rearm batch amount of 0 or less in weapon %s, setting to the default 100 for ballistic primaries.", wip->name);
+			}
+		}
+	//This setting requires a default if not specified, otherwise, rearming will break. - Cyborg17
+	} else {
```

Comment:
This should be `} else if (first_time) {`, or modular tables would have to redefine it every time just to avoid it getting reset to default. Similarly, the above "invalid batch amount" branch will overwrite a (valid) previously-defined rearm ammo increment. I'd suggest, instead, setting the default value to -1 in `weapon_info::reset()` and then do a post-parse "cleanup" of turning that -1 into `REARM_NUM_MISSILES_PER_BATCH` or `REARM_NUM_BALLISTIC_PRIMARIES_PER_BATCH` instead.

Reply:
I think this is what you were suggesting.